alchemist/alchemist-fxui/src/main/kotlin/it/unibo/alchemist/boundary/jfx/events/generic/TriggerAction.kt

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (C) 2010-2020, Danilo Pianini and contributors
+ * listed in the main project's alchemist/build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.boundary.jfx.events.generic
+
+/**
+ * The action that triggers an event dispatcher. May contain information about the trigger.
+ */
+interface TriggerAction
```

Comment:
Why do we need this tag interface?

Reply:
Just to restrict what types can be used in implementations of `EventDispatcher` instead of having `EventDispatcher<T : Any, E : Event>`. I think this also makes the usage of classes like [`KeyboardTriggerAction`](https://github.com/Vuksaa/Alchemist/blob/develop/alchemist/alchemist-fxui/src/main/kotlin/it/unibo/alchemist/boundary/jfx/events/keyboard/KeyboardTriggerAction.kt) more apparent.